pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -321,9 +321,7 @@ func (c *Controller) onServiceEvent(curr interface{}, event model.Event) error {
 			c.nodeSelectorsForServices[svcConv.Hostname] = getNodeSelectorsForService(*svc)
 		}
 		c.servicesMap[svcConv.Hostname] = svcConv
-		if instances == nil {
-			delete(c.externalNameSvcInstanceMap, svcConv.Hostname)
-		} else {
+		if len(instances) > 0 {
```

Comment:
why is this removed? won't we leak now?

Reply:
What i did here is only add to `externalNameSvcInstanceMap` when it is a real externalName service with instances. So absolutely no leak.